### ISSUE TITLE
Docker Container Version

### ISSUE THREAD

**Issue by amouat** [astefanutti/decktape#27]
Hi,

I've made a Dockerized version of Decktape. It's kinda nice, as you don't need to worry about install phantomJS etc. Instead you can just run something like:

    docker run amouat/decktape reveal http://192.168.99.100:8000/ /pres.pdf

And you're at the races. If you like, I'm happy for you to copy the Dockerfile into your project, which has the advantage that you can set-up an automated build that will build a new container whenever the source code changes. Otherwise, I'll just force a rebuild of my project now and again.

https://github.com/amouat/decktape-docker
https://hub.docker.com/r/amouat/decktape/

Adrian.

**Comment by hadim** [comment]
Very cool iniative ! That should be the default way to use decktape :-D

**Comment by astefanutti** [comment]
@amouat awesome! I've just created a Docker Hub repository and setup automated build: https://hub.docker.com/r/astefanutti/decktape/.

There is already a Dockerfile (just init by @ssanderson) though feel free to update it. Beside, that'd be great if you could contribute the usage instruction that you've put together in the README. Let me know if you prefer I deal with it.

**Comment by amouat** [comment]
Was that Dockerfile contributed 14 hours ago? Weird coincidence!

You might want to add the last few lines from my Dockerfile where I chmod the binary and set the ENTRYPOINT and CMD, which makes it a little easier to use.

Thanks for setting up the Hub build.

And yeah, just take the text from my repo, that's totally fine (remember to change the image name in the examples though).

**Comment by ssanderson** [comment]
@amouat am I not `chmod`ing in the current Dockercfile?  If so that's a bug I think.  I thought I had added that but I may have pushed before making that change.  Also :+1: to setting ENTRYPOINT and CMD.

With your example of `docker run amouat/decktape reveal http://192.168.99.100:8000/ /pres.pdf`, I think that will end up writing the resulting pdf into the container filesystem, right?  I imagine that's probably not what most people are going to want.  

I imagine the simplest usage for most folks will be to run with `net=host` (so that localhost resolves to the host machine) and with a host-to-container volume mount.  For example:
```
docker run \
    --net=host \
    --volume=/output:/decktape/output \
    amouat/decktape reveal localhost:8000 /output/pres.pdf
```
This would convert the presentation running on the host machine's `localhost:8000` into a pdf at `/output/pres.pdf` on the host.

**Comment by hadim** [comment]
I will make a PR and also add some instructions to the README.

**Comment by hadim** [comment]
We can continue the discussion there #28

**Comment by astefanutti** [comment]
I've just merged #28. Thank you all!
